### DISCUSSION TITLE
Session duration when the user is not active

### DISCUSSION THREAD

**Discussion by swc-kdzekov** [keycloak/keycloak#14753] [Q&A]
Hello,

I wonder about the session duration in Keycloak when the user is not active  (authenticated user to a resource application that is using KC). Namely, the parameter "**SSO Session Idle**" should regulate that. But what I have noticed is that after this time  exceeds ("SSO Session Idle"), the tokens are invalidated but the session can be revoked by reloading the page (the related session record still exists in KC). This can be done within certain amount of time which is a mystery (unclear) to me. After that time, the session is invalidated in KC, and when I reload the page, it redirects me to the login page as expected.
My question would be: can I really control the time span between "SSO Session Idle" and the time of the complete vanishing of the session in KC when the user is not active? 
Or is it something that is internal to KC, that can not be customised?

Best Regards,
Konstantin

**Comment by jegraham-amplify** [comment]
You may be encountering [this](https://www.keycloak.org/docs/latest/server_admin/#_idle_timeouts_note):

> For idle timeouts, a two-minute window of time exists that the session is active. For example, when you have the timeout set to 30 minutes, it will be 32 minutes before the session expires.This action is necessary for some scenarios in cluster and cross-data center environments where the token refreshes on one cluster node a short time before the expiration and the other cluster nodes incorrectly consider the session as expired because they have not yet received the message about a successful refresh from the refreshing node.
